spring-app/src/main/kotlin/pl/starchasers/up/service/FileService.kt

```diff
@@ -54,81 +50,65 @@ class FileServiceImpl(
     @Transactional
     override fun createFile(
         tmpFile: InputStream,
-        filename: Filename,
-        contentType: ContentType,
-        size: FileSize,
-        user: User?
+        filename: String,
+        contentType: String,
+        size: Long
     ): UploadCompleteResponseDTO {
         val actualContentType = when {
-            contentType.value.isBlank() -> ContentType("application/octet-stream")
-            contentType.value == "text/plain" -> ContentType(
-                "text/plain; charset=" + charsetDetectionService.detect(
-                    tmpFile
-                )
-            )
+            contentType.isBlank() -> "application/octet-stream"
+            contentType == "text/plain" -> "text/plain;charset=" + charsetDetectionService.detect(tmpFile)
             else -> contentType
         }
-        val personalLimit: FileSize = user?.maxTemporaryFileSize
-            ?: FileSize(configurationService.getConfigurationOption(ANONYMOUS_MAX_FILE_SIZE).toLong())
+        val personalLimit: Long = configurationService.getConfigurationOption(ANONYMOUS_MAX_FILE_SIZE).toLong()
 
         if (size > personalLimit) throw FileTooLargeException()
 
-        val key = fileStorageService.storeNonPermanentFile(tmpFile, filename)
+        val key = fileStorageService.storeNonPermanentFile(tmpFile, FileName(filename))
         // TODO check key already used
         val accessToken = generateFileAccessToken()
-        val toDeleteDate = Timestamp.valueOf(LocalDateTime.now().plusDays(1))
-        val fileEntry = FileEntry(
-            0,
-            key,
-            filename,
-            actualContentType,
-            null,
-            false,
-            Timestamp.valueOf(LocalDateTime.now()),
-            toDeleteDate,
-            false,
-            accessToken,
-            size,
-            user
-        )
-
-        fileEntryRepository.save(fileEntry)
-
-        return UploadCompleteResponseDTO(key.value, accessToken.value, toDeleteDate)
+        val toDeleteAt = Instant.now().plus(1, ChronoUnit.DAYS)
+
+        fileEntryRepository.insert {
+            set(it.accessToken, accessToken)
+            set(it.contentType, actualContentType)
+            set(it.createdAt, Instant.now())
+            set(it.encrypted, false)
+            set(it.filename, filename)
+            set(it.key, key.value)
+            set(it.password, null)
+            set(it.size, size)
+            set(it.deleteAt, toDeleteAt)
+        }
```

Comment:
feels a bit weird to have this mapping in a service. Is there any type safety mechanism if one field was forgotten?

Reply:
Not with this api. It would be possible with entities, but we wanted to avoid that.